uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js

```diff
@@ -9,12 +10,453 @@ define([
+     * @return {string}
+     */
+    vm.getLeaveStatusByValue = function (id) {
+      var status = vm.leaveRequestStatuses.find(function (status) {
+        return status.value == id;
+      });
+
+      return status ? status.label : null;
+    };
+
+    /**
+     * Returns the class name for filter navigation when name is given
+     *
+     * @param {string} name - name of the status
+     * @return {string}
+     */
+    vm.getNavBadge = function (name) {
+      switch (name) {
+        case 'approved':
+          return 'badge-success';
+        case 'rejected':
+          return 'badge-danger';
+        case 'cancelled':
+        case 'all':
+          return '';
+        default:
+          return 'badge-primary';
+      }
+    };
+    /**
+     * Returns the username when id is given
+     *
+     * @param {string} id - id of the user
+     * @return {string}
+     */
+
+    vm.getUserNameByID = function (id) {
+      var user = vm.filteredUsers.find(function (data) {
+        return data.contact_id == id;
+      });
+      return user ? user.display_name : null;
+    };
+    /**
+     * Labels the given period according to whether it's current or not
+     *
+     * @param  {AbsencePeriodInstance} period
+     * @return {string}
+     */
+
+    vm.labelPeriod = function (period) {
+      return period.current ? 'Current Period (' + period.title + ')' : period.title;
+    };
+
+    /**
+     * Loads the next page for pagination element based on current page no
+     */
+    vm.nextPage = function () {
+      if (vm.pagination.page < vm.totalNoOfPages()) {
+        vm.refresh(++vm.pagination.page);
+      }
+    };
+
+    /**
+     * Returns an array of a given size
+     *
+     * @param {number} n - no of elements in the array
+     * @return {Array}
+     */
+    vm.range = function (n) {
+      return new Array(n || 0);
+    };
+
+    /**
+     * Refreshes the leave request data
+     *
+     * @param {string} page - page number of the pagination element
+     */
+    vm.refresh = function (page) {
+      page = page ? page : 1;
+      loadAllRequests(page);
+    };
+
+    /**
+     * Refreshes the leave request data and also changes current selected leave status
+     *
+     * @param {string} status - status to be selected
+     */
+    vm.refreshWithFilter = function (status) {
+      vm.filters.leaveRequest.leaveStatus = status;
+      vm.refresh();
+    };
+
+    /**
+     * Calculates the total number of pages for the pagination
+     *
+     * @return {number}
+     */
+    vm.totalNoOfPages = function () {
+      return Math.ceil(vm.leaveRequests.table.total / vm.pagination.size);
+    };
+
+    (function init() {
+      $q.all([
+        loadAbsencePeriods(),
+        loadAbsenceTypes(),
+        loadRegions(),
+        loadDepartments(),
+        loadLocations(),
+        loadLevelTypes(),
+        loadStatuses()
+      ])
+        .then(function () {
+          vm.loading.page = false;
```

Comment:
I dont agree. 
1. We are adding one more promise which is not efficient
2. Normally every line of code is written one after the other unless there is an async call, so this usual. Also if there are 5 lines then we would need to create 5 `then` functions